src/components/NavigationBar.tsx

```diff
@@ -1,155 +1,516 @@
-  };
+const NavigationBar: React.FC<INavigationBarProps> = ({ title, image }) => {
+  const { pathname, push } = useRouter();
+  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState<boolean>(false);
+  const [isSendMenuOpen, setIsSendMenuOpen] = useState<boolean>(false);
+  const [isMobile, setIsMobile] = useState<boolean>(false);
+  const [openSubMenu, setOpenSubMenu] = useState<string | null>(null);
+  const [mounted, setMounted] = useState<boolean>(false);
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
 
-  const handleToggleSubMenu = (event: React.MouseEvent<HTMLElement>) => {
-    event.stopPropagation();
-    if (openSubMenu) {
-      handleCloseSubMenu();
+  const toggleMenu = (menu: string, event?: React.MouseEvent<HTMLElement>) => {
+    if (openSubMenu === menu) {
+      setOpenSubMenu(null);
+      setAnchorEl(null);
     } else {
-      setSubMenuAnchorEl(event.currentTarget);
+      setOpenSubMenu(menu);
+      setAnchorEl(event?.currentTarget || null);
     }
   };
 
-  const handleCloseSubMenu = () => {
-    setSubMenuAnchorEl(null);
-  };
-
-  const handleOpenRitiroSubMenu = (event: React.MouseEvent<HTMLElement>) => {
-    event.stopPropagation();
-    setRitiroSubMenuAnchorEl(event.currentTarget);
+  const handleMenuItemClick = (path: string) => {
+    setIsMobileMenuOpen(false);
+    setIsSendMenuOpen(false);
+    setOpenSubMenu(null);
+    push(path);
   };
 
-  const handleToggleRitiroSubMenu = (event: React.MouseEvent<HTMLElement>) => {
-    event.stopPropagation();
-    if (openRitiroSubMenu) {
-      handleCloseRitiroSubMenu();
-    } else {
-      setRitiroSubMenuAnchorEl(event.currentTarget);
-    }
-  };
+  useEffect(() => {
+    const handleResize = () => setIsMobile(window.innerWidth < 900);
+    handleResize();
+    window.addEventListener("resize", handleResize);
+    setMounted(true);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
 
-  const handleCloseRitiroSubMenu = () => {
-    setRitiroSubMenuAnchorEl(null);
-  };
+  if (!mounted) return null;
 
-  const closeMobileMenu = () => {
-    setIsMobileMenuOpen(false);
-  };
+  const isPathActive = (path: string) => pathname === path;
 
-  const closeSendMenu = () => {
-    setIsSendMenuOpen(false);
-  };
+  const renderMenuItems = (items: MenuItem[], isFirstItem: boolean = false) =>
+    items.map((item, index) => {
+      const isParentActive = isPathActive(item.path);
+      return (
+        <Box
+          key={item.path}
+          sx={{
+            margin: "0 8px",
+            position: "relative",
+            ...(isFirstItem && index === 0 && { marginTop: "50px" }),
+          }}
+        >
+          <Box
+            sx={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "space-between",
+              height: "100%",
+              cursor: "pointer",
+              paddingBottom: 0,
+              padding: isMobile ? "0 8px" : "0 16px", 
```

Comment:
try to use px: isMobile ? 1 : 2 instead of padding: isMobile ? "0 8px" : "0 16px",